stats/stats_test.go

```diff
@@ -380,8 +380,51 @@ func TestHistogramNegativeNumbers(t *testing.T) {
 	}
 }
 
+func TestTransferHistogramWithDifferentScales(t *testing.T)  {
+	tP := []float64{75.}
+	var b bytes.Buffer
+	w := bufio.NewWriter(&b)
+	h1 := NewHistogram(2, 15)
+	h1.Record(30)
+	h1.Record(40)
+	h1.Record(50)
+	h2 := NewHistogram(0, 10)
+	h2.Record(20)
+	h2.Record(90)
+	h1.Print(w, "h1 before merge", tP)
+	h2.Print(w, "h2 before merge", tP)
+	h1.Transfer(h2)
+	h1.Print(w, "merged h2 -> h1", tP)
+	h2.Print(w, "h2 should now be empty", tP)
+	w.Flush()
+	actual := b.String()
+	expected := `h1 before merge : count 3 avg 40 +/- 8.165 min 30 max 50 sum 120
+# range, mid point, percentile, count
+>= 30 < 32 , 31 , 33.33, 1
+>= 32 < 47 , 39.5 , 66.67, 1
+>= 47 <= 50 , 48.5 , 100.00, 1
+# target 75% 47.75
+h2 before merge : count 2 avg 55 +/- 35 min 20 max 90 sum 110
+# range, mid point, percentile, count
+>= 20 < 30 , 25 , 50.00, 1
+>= 90 <= 90 , 90 , 100.00, 1
+# target 75% 90
+merged h2 -> h1 : count 5 avg 46 +/- 24.17 min 20 max 90 sum 230
+# range, mid point, percentile, count
+>= 20 < 32 , 26 , 40.00, 2
+>= 32 < 47 , 39.5 , 60.00, 1
+>= 47 < 62 , 54.5 , 80.00, 1
+>= 77 <= 90 , 83.5 , 100.00, 1
+# target 75% 58.25
+h2 should now be empty : no data
+`
+	if actual != expected {
+		t.Errorf("unexpected:\n%s\tvs:\n%s", actual, expected)
+	}
+}
+
 func TestTransferHistogram(t *testing.T) {
-	tP := []float64{100.} // TODO: use 75 and fix bug
```

Comment:
the TODO may not have been very clear but when using 75% instead of 100% some results didn't match my expectations - basically when using 75% I expect the value given to be == the 3rd point / 4 points

Reply:
Could bug be on CalcPercentile method in stats.go? because When I ran TestHistogram test, CalcPercentile method calculates 775 for 75 percentile however 775 is in range of 80 percentile. As you correct below, I have expected to be min value near target percentile.

Reply:
If you approve that there is a bug on CalcPercentile method, I made solution, I would like to commit. Also, merry Christmas @ldemailly :)

Reply:
yes I expect there is a bug ("fix bug" part) though maybe the bug fix can be a different PR for just that todo/bug fix if you don't spot the bug easily

btw those tests relying on the text output are old, written before there was an Export, new test can be more easily written against that or directly calling CalcPercentile